src/modules/UI/scenes/SendConfirmation/SendConfirmation.ui.js

```diff
@@ -20,86 +20,104 @@ import ABSlider from '../../components/Slider/index.js'
 import Gradient from '../../components/Gradient/Gradient.ui'
 
 import * as UTILS from '../../../utils.js'
-
-import type {GuiWallet, CurrencyConverter} from '../../../../types'
-import type {AbcCurrencyWallet, AbcParsedUri, AbcTransaction} from 'airbitz-core-types'
-import type {SendConfirmationState} from './reducer'
+import type {CurrencyConverter, GuiDenomination} from '../../../../types'
+import type { AbcMetadata } from 'airbitz-core-types'
 
 const DIVIDE_PRECISION = 18
 
-export type Props = {
-  sendConfirmation: SendConfirmationState,
-  abcWallet: AbcCurrencyWallet,
+export type StateProps = {
   nativeAmount: string,
+  publicAddress: string,
+  pending: boolean,
+  keyboardIsVisible: boolean,
+  label: string,
+  primaryDisplayDenomination: GuiDenomination,
+  primaryExchangeDenomination: GuiDenomination,
+  secondaryDisplayCurrencyCode: string,
+  secondaryExchangeCurrencyCode: string,
+  networkFeeOption: string,
+  customNetworkFee: any,
+  networkFee: string,
   errorMsg: string | null,
   fiatPerCrypto: number,
-  guiWallet: GuiWallet,
   currencyCode: string,
-  primaryInfo: FlipInputFieldInfo,
   sliderDisabled: boolean,
-  secondaryInfo: FlipInputFieldInfo,
   currencyConverter: CurrencyConverter
 }
 
 export type DispatchProps = {
-  processParsedUri: (AbcParsedUri) => void,
-  updateSpendPending: (boolean) => void,
-  signBroadcastAndSave: (AbcTransaction) => void
+  updateSpendPending: (boolean) => any,
+  signBroadcastAndSave: () => any,
+  resetFees: () => any,
+  updateAmount: (nativeAmount: string, metadata: AbcMetadata) => any
 }
 
+export type Props = DispatchProps & StateProps
+
 type State = {
-  primaryNativeAmount: string,
-  secondaryNativeAmount: string,
+  nativeAmount: string,
+  secondaryDisplayDenomination: any,
   keyboardVisible: boolean
 }
 
-export default class SendConfirmation extends Component<Props & DispatchProps, State> {
+export default class SendConfirmation extends Component<Props, State> {
   constructor (props: Props & DispatchProps) {
     super(props)
-    const amt = props.sendConfirmation.transaction ? props.sendConfirmation.transaction.nativeAmount : '0'
-
     this.state = {
-      primaryNativeAmount: amt,
-      secondaryNativeAmount: '',
+      nativeAmount: props.nativeAmount,
+      secondaryDisplayDenomination: { multiplier: '1' },
       keyboardVisible: false
     }
   }
 
+  componentWillReceiveProps (nextProps: Props) {
+    if (nextProps['secondaryDisplayCurrencyCode'] !== this.props['secondaryDisplayCurrencyCode']) {
+      this.setState({
+        secondaryDisplayDenomination: UTILS.getDenomFromIsoCode(
+          nextProps['secondaryDisplayCurrencyCode']
```

Comment:
Why are you using named object dereferencing? This doesn't get checked by Flow and it's harder to type and read. use `nextProps.secondaryDisplayCurrencyCode` instead.

Reply:
fixed in https://github.com/Airbitz/edge-react-gui/pull/389